source/common/upstream/upstream_impl.cc

```diff
@@ -173,6 +174,30 @@ Stats::ScopeSharedPtr generateStatsScope(const envoy::config::cluster::v3::Clust
       "cluster.{}.", config.alt_stat_name().empty() ? config.name() : config.alt_stat_name()));
 }
 
+Envoy::Upstream::UpstreamLocalAddressSelectorPtr createUpstreamLocalAddressSelector(
+    const envoy::config::cluster::v3::Cluster& cluster_config,
+    const absl::optional<envoy::config::core::v3::BindConfig>& bootstrap_bind_config) {
+  const std::string empty;
+  const std::string& local_address_selector_name =
+      cluster_config.has_upstream_bind_config() &&
+              !cluster_config.upstream_bind_config().local_address_selector_name().empty()
+          ? cluster_config.upstream_bind_config().local_address_selector_name()
+      : bootstrap_bind_config.has_value() ? bootstrap_bind_config->local_address_selector_name()
+                                          : empty;
+  UpstreamLocalAddressSelectorFactory* local_address_selector =
+      !local_address_selector_name.empty()
+          ? Registry::FactoryRegistry<UpstreamLocalAddressSelectorFactory>::getFactory(
+                local_address_selector_name)
+          : nullptr;
+  if (local_address_selector != nullptr) {
+    return local_address_selector->createLocalAddressSelector(cluster_config,
+                                                              bootstrap_bind_config);
+  } else {
+    return std::make_shared<UpstreamLocalAddressSelectorImpl>(cluster_config,
```

Comment:
We should register the default selector also. The name can be "envoy.upstream.local.address.default.selector"

Reply:
This is what I had [earlier](https://github.com/envoyproxy/envoy/pull/27705/commits/5bfb9cb3e7769ce7072e9b74d86242c02698fb49) but it caused issues with envoy mobile tests for some reason. I'll take another stab at it.